vernac/record.ml

```diff
@@ -525,6 +525,11 @@ let declare_class def cumulative ubinders univs id idbuild paramimpls params uni
       List.map (fun x -> impls @ x) fieldimpls
   in
   let binder_name = Namegen.next_ident_away id (Termops.vars_of_env (Global.env())) in
+  let coers = List.map2 (fun coe pri -> match coe with
+      | Vernacexpr.BackInstance -> Some pri
+      | Vernacexpr.NoInstance -> None)
+      coers priorities
```

Comment:
I think that while you're at it you could merge those two lists as the input of that function.